src/update.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Update as U,
+  Message as Msg,
+  CommonMessageBundle as CommonMsg,
+} from './core/types/typegram'
+
+/**
+ * Update types to narrow Context.
+ * The exact structure of this export may change in the future, or be removed entirely.
+ *
+ * Usage:
+ *
+ * ```TS
+ * interface MyCtx<U extends Update = Update> extends Context<U> {
+ *   session: { count: number }
+ * }
+ *
+ * const actionHandler = (ctx: MyCtx<Update.CallbackQuery>) => {...};
+ * bot.action("forward", actionHandler);
+ *
+ * const textHandler = (ctx: MyCtx<Update.Message<Message.Text>>) => {...};
+ * bot.on("text", textHandler);
+ *
+ * const editedMessageHandler = (ctx: MyCtx<Update.EditedMessage>) => {...};
+ * bot.on("edited_message", editedMessageHandler);
+ * ```
+ * @experimental
+ */
+export namespace Update {
```

Comment:
It's a type-only namespace.

```suggestion
export declare namespace Update {
```